### ISSUE TITLE
[windows] display trouble since b4044ef 

### ISSUE THREAD

**Issue by pyglot** [saulpw/visidata#1409]
**Small description**
I tried vd v2.8 based on an apt package and tables displayed correctly. However it had troubles reading .vds and .rec files, so I tried installing from the github repo's. However now the output looks quite broken. Screenshot below:

**Screenshot**
![image](https://user-images.githubusercontent.com/8835587/175772210-43457254-f019-4730-86c1-1d7fc386c974.png)

**System info**

Host OS is Debian 11 bullseye on WSL. 

I get the same output problems with both Windows Terminal and xterm & uxterm. 

Python version is 3.9.2. 

Output of pip3 freeze:
```
antlr4-python3-runtime==4.9.3
beautifulsoup4==4.9.3
chardet==4.0.0
colorama==0.4.5
cycler==0.11.0
et-xmlfile==1.1.0
fonttools==4.33.3
html5lib==1.1
img2pdf==0.4.0
importlib-metadata==4.11.4
kiwisolver==1.4.3
lxml==4.6.3
Markdown==3.3.7
matplotlib==3.5.2
numpy==1.22.4
olefile==0.46
openpyxl==3.0.10
packaging==21.3
pandas==1.4.2
pikepdf==1.17.3+dfsg
Pillow==9.1.1
PyMuPDF==1.17.4
pyparsing==3.0.9
python-dateutil==2.8.2
pytz==2022.1
six==1.16.0
soupsieve==2.2.1
systemrdl-compiler==1.23.0
visidata==2.9.dev0
webencodings==0.5.1
zipp==3.8.0
```

I have tried installing from github with 
`pip3 install git+https://github.com/saulpw/visidata.git@develop`
and from checked out versions of the repo via
`python3 setup.py install --prefix /home/$user/.local`

The problem persists when I check out old versions (e.g. 2.8), so it seems to be something about my system. Does anyone have a clue as to what might be wrong?

**Comment by saulpw** [comment]
This is due to b4044ef02db08743e89ba8ce8a14b92ee3ab0a15 which improved support for RTL text on several terminal emulators.  Thanks for the repro with xterm, we'll look into it.  In the meantime, is there a more modern terminal on windows you could try?  Perhaps kitty or alacritty?

**Comment by pyglot** [comment]
Indeed the problem disappeared when I reverted b4044ef . Thanks! I may have time tomorrow to check another terminal.

**Comment by pyglot** [comment]
I installed alacritty 0.10.1 for Windows and I'm seeing the same characters "FSI" and "PDI" there. This happens both via WSL (1) to Debian bullseye and Ubuntu 20.04, and also via ssh to a VM (also running bullseye). So this is not specific to WSL or Windows Terminal. I don't see how it would matter but all distributions I have are a bit stripped down (without GUI/WM) so if there's some dependency from something commonly installed on desktop GUI systems, it could be that I'm missing something.

**Comment by pyglot** [comment]
Ok, it looks like alacritty does not support Bi-Directional text yet anyway. 

While RTL is mentioned as supported in kitty, kitty doesn't seem to have a Windows native binary and it requires OpenGL 3.3+ which seems a bit of a pain to get with WSL 1.

After having read and understood a bit more about the underlying issues waiting to be resolved ( #1392  https://github.com/alacritty/alacritty/issues/663 https://github.com/microsoft/terminal/issues/538 ), my thoughts tend towards that it is a bit soon to implement support for RTL as a default, but it would be great to have this as a configuration option. (The default can change when there's a reasonable terminal alternative available on all major platforms, but this display bug will probably never go away).

**Comment by saulpw** [comment]
@pyglot, does the current develop head have issues for you?

**Comment by pyglot** [comment]
Hi, the display issue is fixed at the current develop head 4ab88d52c1db9db15e1613939e35b7794af3f65e. (Unrelated the new head is missing urllib3 in requirements.txt.)

A last thought I have is that there should be a way to query the terminal for the capability to interpret these characters. I'm not sure what can be done in that respect at the moment, but to whomever is interested in this functionality I think asking how this capability can be determined at [the kitty github](https://github.com/kovidgoyal/kitty) might produce some small results in the right direction.
